binary-oracle-pair/program/tests/tests.rs

```diff
@@ -0,0 +1,675 @@
+            &pool.authority,
+            &pool.pool_deposit_account.pubkey(),
+            &user_pass_account.pubkey(),
+            &user_fail_account.pubkey(),
+            &pool.token_pass_mint.pubkey(),
+            &pool.token_fail_mint.pubkey(),
+            &user_account.pubkey(),
+            &spl_token::id(),
+            deposit_amount,
+        )
+        .unwrap()],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[&payer], recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+
+    let user_balance_after = get_token_balance(&mut banks_client, &user_account.pubkey()).await;
+    assert_eq!(user_balance_after, deposit_amount);
+
+    // Check balance of pool deposit account after withdraw
+    let pool_deposit_account_balance_after =
+        get_token_balance(&mut banks_client, &pool.pool_deposit_account.pubkey()).await;
+    assert_eq!(pool_deposit_account_balance_after, 0);
+
+    // Check if program burned PASS and FAIL tokens
+    let user_pass_tokens_after =
+        get_token_balance(&mut banks_client, &user_pass_account.pubkey()).await;
+    assert_eq!(user_pass_tokens_after, 0);
+
+    let user_fail_tokens_after =
+        get_token_balance(&mut banks_client, &user_fail_account.pubkey()).await;
+    assert_eq!(user_fail_tokens_after, 0);
+}
+
+#[tokio::test]
+async fn test_decide() {
+    let mut program_context = program_test().start_with_context().await;
+
+    let pool = TestPool::create();
+
+    pool.init_pool(
+        &mut program_context.banks_client,
+        &program_context.payer,
+        &program_context.last_blockhash,
+    )
+    .await;
+
+    let pool_account_data_before = program_context
+        .banks_client
+        .get_account(pool.pool_account.pubkey())
+        .await
+        .unwrap()
+        .unwrap();
+
+    let pool_data_before = state::Pool::unpack(pool_account_data_before.data.as_slice()).unwrap();
+
+    assert!(pool_data_before.decision.is_none());
+
+    let decision = true;
+
+    program_context
+        .warp_to_slot(pool.mint_end_slot + 1)
+        .unwrap();
+
+    let mut transaction = Transaction::new_with_payer(
+        &[instruction::decide(
+            &id(),
+            &pool.pool_account.pubkey(),
+            &pool.decider.pubkey(),
+            decision,
+        )
+        .unwrap()],
+        Some(&program_context.payer.pubkey()),
+    );
+
+    transaction.sign(
+        &[&program_context.payer, &pool.decider],
+        program_context.last_blockhash,
+    );
+    program_context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+
+    let pool_account_data_after = program_context
+        .banks_client
+        .get_account(pool.pool_account.pubkey())
+        .await
+        .unwrap()
+        .unwrap();
+
+    let pool_data_after = state::Pool::unpack(pool_account_data_after.data.as_slice()).unwrap();
+
+    assert!(pool_data_after.decision.unwrap());
+}
```

Comment:
This should include a positive test for withdrawing after a PASS and FAIL decision to make sure that we can get tokens back.

We can later add the negative tests for trying to deposit after `mint_end` and trying to withdraw during the decide time